orttraining/orttraining/core/framework/module_gradient_graph_builder.cc

```diff
@@ -79,6 +79,9 @@ Status ModuleGradientGraphBuilder::Build(const std::vector<std::vector<int64_t>>
 
   // Build the gradient graph.
   ORT_RETURN_IF_ERROR(BuildGradientGraph());
+  if (!config_.build_gradient_graph) {
+    return Status::OK();
+  }
```

Comment:
Can this test move before the call to `BuildGradientGraph()`? IIUC this would enable us to support inference on ORTModules for which we don't currently support building gradients.

Perhaps there's a reason for this ordering but it's unintuitive (to me at least!), so it'd be worth documenting if so :).

Reply:
Ah, I see now that you escape from `BuildGradientGraph()` halfway through by testing this condition again. Perhaps a good opportunity to split `BuildGradientGraph()` into two functions: `OptimizeInferenceGraph()` and `BuildGradientGraph()`?

Reply:
I think it can be moved before the call to ```BuildGradientGraph()```. I was trying to keep all the graph transformations occurring before the gradient graph building process started intact. That may not be required, I think.

I will move the check before the call to ```BuildGradientGraph()```.

Reply:
Sorry, I did not see the second comment before making my earlier comment.
I think breaking down the function into two seems like a good idea. I'll do that.